File: Harmony/Models/HarmonyModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using Harmony.Util;

namespace Harmony.Models
{
    public class HarmonyModel
    {
        public int Id { get; set; }
        public string? UserId { get; set; }

        public string? FirstChord { get; set; }
        public string? SecondChord { get; set; }
        public string? ThirdChord { get; set; }
        public string? FourthChord { get; set; }
        public string? FifthChord { get; set; }
        public string? SixthChord { get; set; }
        public string? SeventhChord { get; set; }

        public string? EigthChord { get; set; }
        public int? NumberOfChords { get; set; }

        public string? Key { get; set; }

        public bool? IsMagic { get; set; }

        
    }
}

File: Harmony/Angular-Harmony/src/app/my-favourites/harmony.model.ts
export interface harmonyCreationDTO{
  firstChord: string;
  secondChord: string;
  thirdChord: string;
  fourthChord: string;
  fifthChord: string;
  sixthChord: string;
  seventhChord: string;
  eigthChord: string;
  numberOfChords: number;
  key: string;
  isMagic: boolean;
}

export interface harmonyDTO{
  id: number;
  userId: string;
  firstChord: string;
  secondChord: string;
  thirdChord: string;
  fourthChord: string;
  fifthChord: string;
  sixthChord: string;
  seventhChord: string;
  eigthChord: string;
  numberOfChords: number;
  key: string;
  isMagic: boolean;
}

export class harmonyVM {

  id: number;
  numberOfChords: number;
  key: string;
  isMagic: boolean;
  chords: string[];

  constructor(harmony:harmonyDTO) {
    this.id = harmony.id;
    this.numberOfChords = harmony.numberOfChords;
    this.key = harmony.key;
    this.isMagic = harmony.isMagic;
    this.chords = [harmony.firstChord, harmony.secondChord, harmony.thirdChord,harmony.fourthChord,harmony.fifthChord,harmony.sixthChord,harmony.seventhChord,harmony.eigthChord]

    this.chords = this.chords.filter( n => n!= null);
  }

}


export class GuitarChord {
  key: string;
  positions: string;
  fingers: string;
  variation: number;

  constructor(key:string, positions:string, fingers:string, variation:number){
    this.key =key;
    this.positions=positions;
    this.fingers = fingers;
    this.variation = variation;
  }
};

export class GuitarHarmony{
  id:number;
  harmonies: GuitarChord[] = [];

  constructor(id:number){this.id = id}

  addHarmony(guitarChord:GuitarChord){
    this.harmonies.push(guitarChord);
  }
}
